### DISCUSSION TITLE
Keep center point during zoom?

### DISCUSSION THREAD

**Discussion by trurl-master** [ValentinH/react-easy-crop#202] [Q&A]
@ValentinH Hey hi! :)

I find it frustrating when zooming without scroll (using buttons or slider) that the center point shifts towards the center of the image (check any of the examples to see what I'm talking about). Is there a way to fix it? Or maybe we should modify the default behavior and keep the center point in place?

Thanks for the great lib!

**Comment by ValentinH** [comment]
Thanks for the question.

Would you expect the zoom to use the current center of the crop view?
The way it's done when zooming with the mouse is to also translate the image to follow the cursor.
When zooming without the mouse, it means that the crop position should also be adjusted to take the new zoom value into account. This can be done in the parent component but it might be tricky.

**Reply by trurl-master** [reply]
Thanks for the answer.

Yeah, as I understand `zoom` is just a parameter to `scale` inside the CSS transform property, so the crop center should also be translated to compensate for the displacement. I tried to do that outside the component, but it feels dirty :)

I think it feels natural to zoom towards the current center instead of to some seemingly arbitrary point on the image. It's done perfectly when zooming with the scroll and feels weird when using buttons or some other way. Especially when you need to suddenly change the zoom by a large amount and your crop area is far from the center of the image, it seems like it just jumps to some seemingly random spot.

**Reply by ValentinH** [reply]
I'm not sure this can be done within the library itself: the `crop` and `zoom` property are controlled by the parent component. If we were to compute a new `crop` when the `zoom` prop change, it will conflict with the current behaviour of  following the mouse when zooming with scroll. There is actually also a feature to have nice zooming with mobile gestures, this would also break it.
